galaxy_ng/app/api/v3/viewsets/namespace.py

```diff
@@ -52,3 +55,21 @@ def get_permissions(self):
             permission_list.append(permissions.IsNamespaceOwnerOrPartnerEngineer())
 
         return permission_list
+
+    @transaction.atomic
+    def create(self, request, *args, **kwargs):
+        """Override to also create inbound pulp repository and distribution."""
+        name = INBOUND_REPO.format(namespace_name=request.data['name'])
+        repo = AnsibleRepository.objects.create(name=name)
+        AnsibleDistribution.objects.create(name=name, base_path=name, repository=repo)
+        return super().create(request, *args, **kwargs)
+
+    @transaction.atomic
+    def destroy(self, request, *args, **kwargs):
+        """Override to also delete inbound pulp repository and distribution.
+        RepositoryVersion objects get deleted on delete of AnsibleRepository.
+        """
+        name = INBOUND_REPO.format(namespace_name=kwargs['name'])
+        AnsibleRepository.objects.get(name=name).delete()
```

Comment:
This code doesn't handle NotFound exception. IIRC it will end up with 500 response.

Reply:
Updated